File: python/1193_conversao_entre_bases.py
n = int(input())
for i in range(1,n+1):
    num, type = map(str, input().split())
    print("Case {}:" .format(i))
    if type == bin:
        print("{} dec" .format(int("0b" + num)))
        print("{} hex" .format(hex(int(num))))
    elif type == "dec":
        num = int(num)
        print("{} hex" .format(hex(num)))
        print("{} bin" .format(bin(num)))
    else:
        num = int(num)
        print("{} dec" .format(int(num)))
        print("{} bin" .format(bin(num)))
        

File: C, C++/1193_conversao_entre_bases.cpp
#include <iostream>
#include <sstream>
#include <stdlib.h>
#include <stdio.h>
using namespace std;

int main(){
    int n, i, con;
	string type, num;
	cin >> n;
	for(i = 0;i < n; i++){
	    cin >> num >> type;
		stringstream stream;
		printf("Case %d:\n", i);
		if(type == "bin"){
			stream << num;
			stream >> con;
			printf("%d dec\n", con);
			printf("%X hex\n", con);
		}else if(type == "dec"){
		    stream << num;
			stream >> con;
			printf("%X hex\n", con);
			printf("%d bin\n", con);
		}else{
			stream << num;
			stream >> hex >> con;
			printf("%d dec\n", con);
			printf("%d bin\n", con);
		}
	}
    
return 0;
}
